gobblin-metrics/src/main/java/gobblin/metrics/MetricContext.java

```diff
@@ -171,6 +200,25 @@ public void addChildContext(String childContextName, MetricContext childContext)
     return this.reportFullyQualifiedNames ? super.getNames() : getSimpleNames();
   }
 
+
+  public void sendEvent(GobblinTrackingEvent event) {
+    GobblinTrackingEvent eventCopy = GobblinTrackingEvent.newBuilder(event).build();
```

Comment:
Per our discussion offline, please make it clear in the Javadoc that `event` is not supposed to be reused.